pkg/cli/admin/upgrade/status/alerts.go

```diff
@@ -0,0 +1,73 @@
+package status
+
+import (
+	"time"
+)
+
+type AlertLabels struct {
+	AlertName string `json:"alertname,omitempty"`
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+	Reason    string `json:"reason,omitempty"`
+	Severity  string `json:"severity,omitempty"`
+}
+
+type AlertAnotations struct {
+	Description string `json:"description,omitempty"`
+	Summary     string `json:"summary,omitempty"`
+}
+
+type Alert struct {
+	Labels      AlertLabels     `json:"labels,omitempty"`
+	Annotations AlertAnotations `json:"annotations,omitempty"`
+	State       string          `json:"state,omitempty"`
+	Value       string          `json:"value,omitempty"`
+	ActiveAt    time.Time       `json:"activeAt,omitempty"`
+	Level       string          `json:"partialResponseStrategy,omitempty"`
+}
+
+type AlertData struct {
+	Status string `json:"status"`
+	Data   Data   `json:"data"`
+}
+
+type Data struct {
+	Alerts []Alert `json:"alerts"`
+}
+
+func parseAlertDataToInsights(alertData AlertData, startedAt time.Time) []updateInsight {
+	var alerts []Alert = alertData.Data.Alerts
+	var updateInsights []updateInsight
+
+	for _, alert := range alerts {
+		if startedAt.After(alert.ActiveAt) {
+			continue
+		}
+		updateInsights = append(updateInsights, updateInsight{
+			startedAt: alert.ActiveAt,
+			impact: updateInsightImpact{
+				level:      alertResponseStrategyToImpact(alert.Level),
```

Comment:
I'm not sure what `partialResponseStrategy` is about (and that's what you're currently mapping to `Level`).  But for `impactLevel` here, I think you want to consume `alert.Labels.Severity`:

```console
$ OC_ENABLE_CMD_INSPECT_ALERTS=true oc adm inspect-alerts | jq '.data.alerts[].labels.severity' | sort | uniq -c
      9 "critical"
      2 "info"
      1 "none"
     26 "warning"
```

Without teaching this function alert-specific issues, I'd expect something like the following mapping:

* `none` (this is `Watchdog`, and we don't need to pass that along, although we might want to warn when it isn't present?).
* `info`, map to `infoImpactLevel`.
* `warning`, map to `warningImpactLevel`.
* `critical`, map to `errorImpactLevel`.  Seems unlikely that anything that's only coming in through alerts deserves `criticalInfoLevel`, and if we decide it does, we probably want to hand-pick those alert names.  For example:

```console
$ OC_ENABLE_CMD_INSPECT_ALERTS=true oc adm inspect-alerts | jq -c '.data.alerts[] | select(.labels.severity == "critical") | {alertname: .labels.alertname, annotations}' | sort | uniq -c
      9 {"alertname":"promtail-dropping-data","annotations":{"message":"Promtail is unable to ship logs and dropping data."}}
```

is currently firing in build02, and while that is claiming it is `critical`, it is unlikely to actually matter vs. cluster updates.